### PR TITLE
render crosses

### LINKED ISSUE #1060: Add rendering for man_made=cross
related to #131

### PR DESCRIPTION
**Pull request by matkoniecz**
add rendering for historic=wayside_cross - fixes part of #131
add rendering for man_made=cross - fixes #1060 

https://cloud.githubusercontent.com/assets/899988/8518992/67569bf4-23cf-11e5-9331-1b84eb4c1e64.png
https://cloud.githubusercontent.com/assets/899988/8518993/6758ccd0-23cf-11e5-85ef-9ce6fa8e1bdc.png
https://cloud.githubusercontent.com/assets/899988/8518995/67594f7a-23cf-11e5-9eba-5d77b7396abd.png
https://cloud.githubusercontent.com/assets/899988/8518994/67591d84-23cf-11e5-8dfb-19fca79fa874.png
https://cloud.githubusercontent.com/assets/899988/8518997/675c9be4-23cf-11e5-8b1d-23280caea8f2.png
https://cloud.githubusercontent.com/assets/899988/8518996/6759b582-23cf-11e5-81aa-78d6b1bcde9a.png
https://cloud.githubusercontent.com/assets/899988/8518998/6770018e-23cf-11e5-90be-fe77a8597438.png
https://cloud.githubusercontent.com/assets/899988/8519001/67730582-23cf-11e5-89c8-3dacab2d785f.png
https://cloud.githubusercontent.com/assets/899988/8518999/6770b296-23cf-11e5-8a18-dacb82e09a12.png
https://cloud.githubusercontent.com/assets/899988/8519002/677366bc-23cf-11e5-916c-0d0d4ebe9d2a.png
https://cloud.githubusercontent.com/assets/899988/8519000/67727338-23cf-11e5-9c86-eb6f9567923d.png
https://cloud.githubusercontent.com/assets/899988/8519003/6775f9d6-23cf-11e5-9f33-febabebaa609.png
https://cloud.githubusercontent.com/assets/899988/8519004/6786a7a4-23cf-11e5-9839-f1ec60998a4d.png
https://cloud.githubusercontent.com/assets/899988/8519006/678e0e0e-23cf-11e5-8028-ea1c32ec58aa.png

### CODE REVIEW

**Comment by kocio-pl** [comment]
Would be great!

**Comment by matkoniecz** [comment]
> render orthodox crosses in a different way

That would be a good idea, and there is even issue for that - but for now even amenity=place_of_worship has no special rendering.

**Comment by math1985** [comment]
I think rendering this from z14 is way too early, I'd rather start rendering these from z16 or even z17.

**Comment by kocio-pl** [comment]
On the other hand they can be important landmarks in rural areas, which makes them more important than only for micromapping purposes, plus they are probably always alone, so they wouldn't pollute the map anyway.

**Comment by matkoniecz** [comment]
@math1985 

z14 is only for named ones - and it is in low priority layer.  In z15 also unnamed appear but still in low priority layer. 

Given that on z16 gates appear (see for an example http://www.openstreetmap.org/#map=16/50.0261/19.9284 ) I think that z17 is an overkill.

I may make test rendering for an additional locations - at least in places that I know rendering seemed to be OK.

**Comment by matkoniecz** [comment]
rebased

**Comment by math1985** [comment]
Density of named crosses is very high in some regions (for example Saarland) so I still think the zoomlevels need to go up.

**Comment by kocio-pl** [comment]
Are those crosses in Saarland named?

**Comment by matkoniecz** [comment]
> Density of named crosses is very high in some regions (for example Saarland) so I still think the zoomlevels need to go up.

This batch of examples is from Saarland. I think that zoomlevels are OK. I think that appearing about at the same zlevel as highway=path is reasonable, but moving appearance of unnamed oned a bit later also would be OK.

https://cloud.githubusercontent.com/assets/899988/8912740/1476fcfe-3495-11e5-93ba-f7a14a421708.png
https://cloud.githubusercontent.com/assets/899988/8912742/149ed54e-3495-11e5-8bea-86d16aee1f23.png
https://cloud.githubusercontent.com/assets/899988/8912745/14a0bda0-3495-11e5-9437-9e98d6b357d6.png
https://cloud.githubusercontent.com/assets/899988/8912744/14a04e7e-3495-11e5-8e6d-43606c9a77e3.png
https://cloud.githubusercontent.com/assets/899988/8912743/149ef2ae-3495-11e5-9a8c-d0560c0da40c.png
https://cloud.githubusercontent.com/assets/899988/8912746/14a2085e-3495-11e5-814f-ef7e98cdeef7.png
https://cloud.githubusercontent.com/assets/899988/8912741/148d4b4e-3495-11e5-8320-ab07fa095378.png
https://cloud.githubusercontent.com/assets/899988/8912747/14aa6e22-3495-11e5-9af7-105905924ff1.png
https://cloud.githubusercontent.com/assets/899988/8912748/14b902d4-3495-11e5-9294-2cea639c3ffa.png
https://cloud.githubusercontent.com/assets/899988/8912750/14bbd270-3495-11e5-938c-3a9cd1b4f4e5.png
https://cloud.githubusercontent.com/assets/899988/8912749/14bb1fa6-3495-11e5-9d51-882ef7f51279.png
https://cloud.githubusercontent.com/assets/899988/8912752/14be65f8-3495-11e5-8a7a-1419a94ce31f.png

**Comment by kocio-pl** [comment]
I see no problems on all of these samples.

**Comment by matkoniecz** [comment]
@math1985 

> Density of named crosses is very high in some regions (for example Saarland) so I still think the zoomlevels need to go up.

Can you look at https://github.com/gravitystorm/openstreetmap-carto/pull/1636#issuecomment-125280630 ? Importance of crosses (as orientation point, religious function, etc) probably differs significantly from country to country so feedback would be really useful.

**Comment by pnorman** [comment]
I'm not sure we should be rendering these at all, and don't think the icon proposed is clear in meaning.

**Comment by kocio-pl** [comment]
> I'm not sure we should be rendering these at all, and don't think the icon proposed is clear in meaning.

IMO it's hard to find something more sticking to the ground truth: the shape is obvious, the zoom level and size suggest something smaller+less important than religious buildings. I can't even guess what could be misleading in this case.

**Comment by mboeringa** [comment]
> IMO it's hard to find something more sticking to the ground truth

Yes, I can only think of one obvious addition that might help recognition: give the cross a small pedestal of a few pixels wide and high. This would distinguish it a little more of the standard cross icon.

**Comment by kocio-pl** [comment]
There are some crosses which are memorials at the same time, [like this for example](http://www.openstreetmap.org/way/238663498). I think they should be rendered as crosses rather than memorials, because we have a lot of different types, excluding only monumental ones, but they all look the same. What do you think about this problem?

**Comment by math1985** [comment]
Aren't all crosses memorials - for Jezus Christ? :)

**Comment by kocio-pl** [comment]
Right. :wink: However I'd like to know also if there are no technical obstacles to make property of being cross priority over being memorial?

**Comment by matkoniecz** [comment]
@kocio-pl 

> However I'd like to know also if there are no technical obstacles to make property of being cross priority over being memorial?

No, it is quite easy - just modify order in coalesce queries like it is done in #1421

**Comment by kocio-pl** [comment]
M-kay, yet another thing to learn... However this one sounds rather easy.

**Comment by matkoniecz** [comment]
> I'm not sure we should be rendering these at all, and don't think the icon proposed is clear in meaning.

> Particularly around other religious objects, e.g. https://cloud.githubusercontent.com/assets/899988/8519006/678e0e0e-23cf-11e5-8028-ea1c32ec58aa.png

Hm. It may be another case of cultural differences. In Poland crosses are frequently mapped.

But maybe there is distinction on these map that I missed here. @pnorman - are you familiar with maps that show crosses? (if yes - then I should investigate why my rendering is not working, if no - then it is cultural difference about what may appear on the map).

**Comment by BalooUriza** [comment]
On Mon, Aug 31, 2015 at 1:20 PM, Mateusz Konieczny <notifications@github.com
> wrote:

> I'm not sure we should be rendering these at all, and don't think the icon
> proposed is clear in meaning.
>
> Particularly around other religious objects, e.g.
> https://cloud.githubusercontent.com/assets/899988/8519006/678e0e0e-23cf-11e5-8028-ea1c32ec58aa.png
>
> Hm. It may be another case of cultural differences. In Poland crosses are
> frequently mapped.
>
> But maybe there is distinction on these map that I missed here. @pnorman
> <https://github.com/pnorman> - are you familiar with maps that show
> crosses? (if yes - then I should investigate why my rendering is not
> working, if no - then it is cultural difference about what may appear on
> the map)
>
I could see how including crosses could be helpful.  MDT in Montana, for
example, posts steel crosses on the highway shoulder closest to the point
of impact or at the location of roadway departure of fatal traffic
incidents, one per fatality.  The officially stated purpose is as a
reminder to not drive like a nutter.  However, they also tend to serve as a
landmark for landmark-based navigation.

**Comment by dieterdreist** [comment]
sent from a phone

> Am 01.09.2015 um 13:26 schrieb BalooUriza <notifications@github.com>:
> 
> I could see how including crosses could be helpful. MDT in Montana, for
> example, posts steel crosses on the highway shoulder closest to the point
> of impact or at the location of roadway departure of fatal traffic
> incidents, one per fatality. The officially stated purpose is as a
> reminder to not drive like a nutter. However, they also tend to serve as a
> landmark for landmark-based navigation.


yes, this kind of crosses are common in Germany and Italy as well. I believe they are usually set up by the relatives. Wouldn't see them as landmarks though, as they're normally quite small (ca. 50cm / 2ft). Maybe we should have a subtag or different main tag to distinguish them from other kind of crosses?

cheers 
Martin

**Comment by BalooUriza** [comment]
On Wed, Sep 2, 2015 at 6:35 AM, dieterdreist <notifications@github.com>
wrote:

>
>
> sent from a phone
>
> > Am 01.09.2015 um 13:26 schrieb BalooUriza <notifications@github.com>:
> >
> > I could see how including crosses could be helpful. MDT in Montana, for
> > example, posts steel crosses on the highway shoulder closest to the point
> > of impact or at the location of roadway departure of fatal traffic
> > incidents, one per fatality. The officially stated purpose is as a
> > reminder to not drive like a nutter. However, they also tend to serve as
> a
> > landmark for landmark-based navigation.
>
>
> yes, this kind of crosses are common in Germany and Italy as well. I
> believe they are usually set up by the relatives. Wouldn't see them as
> landmarks though, as they're normally quite small (ca. 50cm / 2ft). Maybe
> we should have a subtag or different main tag to distinguish them from
> other kind of crosses?
>

I haven't tagged any myself (the montana ones are the only ones I see
consistently placed, and being made out of slats of scrap steel and
painted, aren't generally visible from aerial photography unless placed
perpendicular to the angle of the sun with the aerial taken early morning
or late afternoon), but I'd be inclined to tag it as historic=wayside_cross
in the Montana instance.  Also, appears I was wrong, it's the American
Legion that places them, though MDT does endorse the practice and has
official signage explaining their meaning (the line indicating who
maintains them is nearly invisible when driving, as they tend to be placed
at the start of high incident corridors where you're already (in theory)
more focused on the road than what's next to it).
http://mtlegion.org/department/highway-fatality-markers.html

**Comment by matkoniecz** [comment]
rebased

**Comment by math1985** [comment]
Render crosses from z14 is a dealbreaker for me - in rural areas it might work, but the results are very confusing in cities, where we currently already suffer from too high icon density. And rendering crosses so much than shops does not make sense to me.
@matkoniecz Any compromis we can settle on? :)

**Comment by matkoniecz** [comment]
> Render crosses from z14 is a dealbreaker for me

I updated PR to stop also named crosses from appearing at z14.

File: project.yaml [hunk 1]
```diff
@@ -1650,7 +1650,9 @@ Layer:
               'historic_' || CASE WHEN historic IN ('memorial', 'monument', 'archaeological_site') THEN historic ELSE NULL END,
               'highway_'|| CASE WHEN highway IN ('bus_stop', 'elevator', 'traffic_signals', 'ford') THEN highway ELSE NULL END,
               'power_' || CASE WHEN power IN ('generator') THEN power ELSE NULL END,
-              'tourism_' || CASE WHEN tourism IN ('viewpoint') THEN tourism ELSE NULL END
+              'tourism_' || CASE WHEN tourism IN ('viewpoint') THEN tourism ELSE NULL END,
+              'man_made_' || CASE WHEN man_made IN ('cross') THEN man_made ELSE NULL END,
+              'historic_' || CASE WHEN historic IN ('wayside_cross') THEN historic ELSE NULL END
```
Comment (math1985) [inline project.yaml]: @matkoniecz Did you deliberately add these as separate lines?

Comment (matkoniecz) [inline project.yaml]: Yes, it was intentional to render [tourism=viewpoint, man_made=cross] as viewpoint rather than as a cross.

**Comment by math1985** [comment]
Merged - but as far as I'm concerned, crosses could still go up in zoomlevel.

**Comment by kocio-pl** [comment]
@matkoniecz I've just noticed, that you have added new lines for historic and man made objects, while you could just add crosses at the end of existing lines. Could you explain why did you do it?

**Comment by matkoniecz** [comment]
Main difference is how for example historic=monument + man_made=cross and similar would be rendered with monument icon rather than cross icon.

Other likely combinations include tourism=viewpoint.

It is also possible that it is simply a mistake.

In both cases I am not sure whatever separate lines are worth keeping.

**Comment by kocio-pl** [comment]
Thanks for a feedback, I will check that later.

**Pull request closed without merging**